homu/server.py

```diff
@@ -698,31 +706,43 @@ def report_build_res(succ, url, builder, state, logger, repo_cfg):
                 ))
                 state.change_labels(LabelEvent.SUCCEED)
 
-                def set_ref():
+                def set_ref_inner():
                     utils.github_set_ref(state.get_repo(), 'heads/' +
                                          state.base_ref, state.merge_sha)
                     if state.test_on_fork is not None:
                         utils.github_set_ref(state.get_test_on_fork_repo(),
                                              'heads/' + state.base_ref,
                                              state.merge_sha, force=True)
-                try:
+
+                def set_ref():
                     try:
-                        set_ref()
+                        set_ref_inner()
                     except github3.models.GitHubError:
                         utils.github_create_status(
                             state.get_repo(),
                             state.merge_sha,
                             'success', '',
                             'Branch protection bypassed',
                             context='homu')
-                        set_ref()
+                        set_ref_inner()
 
-                    state.fake_merge(repo_cfg)
+                error = None
+                for i in range(0, 5):
+                    try:
+                        set_ref()
+                        state.fake_merge(repo_cfg)
```

Comment:
Shouldn't `error` be reset to `None` if `set_ref` succeeds?

Reply:
Yes, good catch.